File: baseAdapter/src/main/java/com/meicet/adapter/databinding/adapter/ViewAdapter.java
package com.meicet.adapter.databinding.adapter;

import android.graphics.drawable.Drawable;
import android.graphics.drawable.LevelListDrawable;
import android.view.View;

import androidx.databinding.BindingAdapter;

import com.meicet.adapter.utils.ViewExtensionKt;

public class ViewAdapter {

    @BindingAdapter("gone")
    public static void bindGone(View view, boolean gone) {
        ViewExtensionKt.gone(view, gone);
    }

    @BindingAdapter("visiable")
    public static void bindViewVisibility(View view, boolean visiable) {
        ViewExtensionKt.visiable(view, visiable);
    }

    @BindingAdapter("levelBG")
    public static void bindLevelBG(View view, int level) {
        Drawable data = view.getBackground();
        if (data instanceof LevelListDrawable) {
            data.setLevel(level);
        }
    }

    @BindingAdapter("isSelect")
    public static void bindSelect(View view, boolean isSelect) {
        view.setSelected(isSelect);
    }

    @BindingAdapter("onClickSelect")
    public static void bindOnClickSelect(View view, boolean isSelect) {
        view.setSelected(isSelect);
        view.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                v.setSelected(!v.isSelected());
            }
        });
    }

    @BindingAdapter("onClickOnce")
    public static void bindOnClickOnce(View view, View.OnClickListener onClickListener) {
        ViewExtensionKt.onClickOnce(view, onClickListener);
    }

    @BindingAdapter("onClickSingle")
    public static void bindOnClickSingle(View view, View.OnClickListener onClickListener) {
        ViewExtensionKt.onClickOnceSingle(view, onClickListener);
    }
}

File: baseAdapter/src/main/java/com/meicet/adapter/databinding/adapter/ViewAdapter.kt
package com.meicet.adapter.databinding.adapter

import android.view.View
import androidx.databinding.BindingAdapter
import com.meicet.adapter.utils.gone
import com.meicet.adapter.utils.onClickOnce
import com.meicet.adapter.utils.onClickOnceSingle
import com.meicet.adapter.utils.visiable


/**
 * author : yyl on 2020/4/1
 * desc:
 */
object ViewAdapter {
    @JvmStatic
    @BindingAdapter("gone")
    fun bindGone(view: View, gone: Boolean) {
        view.gone(gone)
    }

    @JvmStatic
    @BindingAdapter("visibility")
    fun bindVisibility(view: View, visibility: Boolean) {
        view.visiable(visibility)
    }
    @JvmStatic
    @BindingAdapter("visiable")
    fun bindViewVisibility(view: View, visiable: Boolean) {
        view.visiable(visiable)
    }
    @JvmStatic
    @BindingAdapter("isSelect")
    fun bindSelect(view: View, isSelect: Boolean) {
        view.isSelected = isSelect
    }
    @JvmStatic
    @BindingAdapter("onClickSelect")
    fun bindOnClickSelect(view: View, isSelect: Boolean) {
        view.isSelected = isSelect
        view.setOnClickListener {
            it.isSelected = !it.isSelected
        }
    }
    @JvmStatic
    @BindingAdapter("onClickOnce")
    fun bindOnClickOnce(view: View, onClickOnce: View.OnClickListener) {
        view.onClickOnce {
            onClickOnce.onClick(it)
        }
    }
    @JvmStatic
    @BindingAdapter("onClickSingle")
    fun bindOnClickSingle(view: View, onClickSingle: View.OnClickListener) {
        view.onClickOnceSingle {
            onClickSingle.onClick(it)
        }
    }


}